### ISSUE TITLE
unused variable for function is using decorator

### ISSUE THREAD

**Issue by lovedota** [palantir/tslint#667]
Hi 
I got an trouble while try to add decorator for a private function in *class*, This is the store in ReactJS, Could you please ignore the warning for the function has decorator ? Thanks !

``` ts
class TodoStore extends BaseStore {
  @handle(TodoConstants.TODO_LOAD_COMPLETE)
  private convertTodosToViewModel(action: TodoAction) {
  }
```

**Comment by JKillian** [comment]
It seems that might be too broad of a change; that is, there would be lots of cases where somebody might be using a decorator and still want to check for unused variables

**Comment by adidahiya** [comment]
This seems like a good use case for the inline enable/disable TSLint comments.

If you want a more broadly applicable change to the way `no-unused-variable` works, please write up a more detailed proposal for it.
